pkg/kubectl/cmd/config/config.go

```diff
@@ -429,6 +430,10 @@ func getConfigFromFileOrDie(filename string) *clientcmdapi.Config {
 		return clientcmdapi.NewConfig()
 	}
 
+	if config.Clusters == nil {
+		config.Clusters = map[string]*clientcmdapi.Cluster{}
```

Comment:
`clientcmdapi.NewConfig()` appears to be setting `config.Clusters` to a non-nil value; I wonder if `clientcmdlatest.Codec.DecodeInto()` is unsetting it?